File: GetResourceGroup.cs
// Get Azure existing Resource Group
// https://www.pulumi.com/ai/conversations/5812c0af-5b14-415b-b689-87dc5b39896c

using Pulumi;
using Pulumi.AzureNative.Resources;
using System.Collections.Generic;
using System.Threading.Tasks;

return await Deployment.RunAsync(async () =>
{
    var existingResourceGroup = await GetResourceGroup.InvokeAsync(new GetResourceGroupArgs
    {
        ResourceGroupName = "my-existing-resource-group", // Replace with your resource group name
    });

    return new Dictionary<string, object?>
    {
        ["resourceGroupName"] = existingResourceGroup.Name,
        ["location"] = existingResourceGroup.Location
    };
});

File: GetResourceGroup.fs
// Get Azure existing Resource Group in F#
// https://www.pulumi.com/ai/conversations/dc60e9ba-daaa-451f-a6ae-ebe60ce3b14c

open Pulumi
open Pulumi.AzureNative.Resources

let getResourceGroup resourceGroupName =
    async {
        let! rg = GetResourceGroup.InvokeAsync (GetResourceGroupArgs(ResourceGroupName = resourceGroupName))
        return rg
    }

let deployment = Deployment.runAsync (fun () ->
    async {
        let! rg = getResourceGroup "your-resource-group-name" // Replace with the actual resource group name
        // Outputs resource group details
        return dict [ ("resourceGroupName", Output<string>.Create(rg.Name)) ]
    }
) :> _

deployment
